sim/src/test/scala/bridges/BridgeSuite.scala

```diff
@@ -78,15 +72,69 @@ abstract class BridgeSuite(
   mkdirs()
   behavior.of(s"$targetName")
   elaborateAndCompile()
-  compileMlSimulator(backendSimulator)
-  runTest(backendSimulator)
+  for (backend <- Seq("vcs", "verilator")) {
+    compileMlSimulator(backend)
+    runTest(backend)
+  }
 }
 
+class UARTTest(targetConfig: String) extends BridgeSuite("UARTModule", "UARTConfig", targetConfig) {}
+
 class UARTF1Test    extends BridgeSuite("UARTModule", "UARTConfig", "DefaultF1Config")
 class UARTVitisTest extends BridgeSuite("UARTModule", "UARTConfig", "DefaultVitisConfig")
 
+class BlockDevTest(targetConfig: String) extends BridgeSuite("BlockDevModule", "BlockDevConfig", targetConfig) {
+  override def runTest(backend: String, debug: Boolean, shouldPass: Boolean) {
+    val prefix     = if (shouldPass) "pass in " else "fail in "
```

Comment:
See #1336, it will simplify it. I am not at all happy with the pattern you've shown me since it intermixes make calls with test cases. I think the whole mechanism should be reconsidered, but there is a limit on how much time I can spend on this. After #1335 and #1336 tests will work and after library separation a lot of stuff will be simplified. Please consider a rewrite then, but now I cannot put more time into this.